base/linalg/dense.jl

```diff
@@ -291,7 +291,7 @@ julia> kron(A, B)
 ```
 """
 function kron{T,S}(a::AbstractMatrix{T}, b::AbstractMatrix{S})
-    R = Array{promote_type(T,S)}(size(a,1)*size(b,1), size(a,2)*size(b,2))
+    R = Array{promote_op(*,T,S)}(size(a,1)*size(b,1), size(a,2)*size(b,2))
```

Comment:
If we could avoid calling `promote_op` here (which is, of course, better than `promote_type` here), and determine the output element type based on the types of the actually computed values, that would be great. Maybe some fancy comprehension? But that's probably outside the scope of this PR.